lib/galaxy/util/magic_numbers/__init__.py

```diff
@@ -0,0 +1,98 @@
+import os
+import io
+
+# The number of bytes to be read from a file/stream when searching for a
+# magic number.  No magic number is even close to this long, but given most
+# block devices read 4K chunks this is likely a moot point.
+BLOCK_SIZE=1024
+
+
+class MagicNumber:
+    '''
+    Information about magic numbers loaded from the tsv and text data files.
+    '''
+    def __init__(self, number, extension, description):
+        self.number = number
+        self.bytes = [ int(hex, 16) for hex in number.split() if hex != '??']
+        self.extension = extension.strip()
+        self.description = description.strip()
+
+
+class TrieNode:
+    '''
+    A node in the search tree.
+
+    If the formats list is not empty then the sequence of bytes that led to this
+    node is a known magic number.
+    '''
+    def __init__(self):
+        self.children = [None] * 256
+        self.formats = []
+
+
+class MagicNumberSniffer:
+    '''
+    The prefix search tree (trie) used to identify magic numbers at
+    the start of a file.
+    '''
+    def __init__(self):
+        self.root = TrieNode()
+        this_dir = os.path.dirname(__file__)
+        with open(os.path.join(this_dir, "magic-numbers.tsv")) as fp:
+            for line in fp.readlines():
+                parts = line.split('\t')
+                number = MagicNumber(*parts)
+                self.add(number)
+        with open(os.path.join(this_dir, "magic-numbers.txt")) as fp:
+            for line in fp.readlines():
+                parts = line.split('\t')
+                bytes = ' '.join([ hex(ord(c)).replace('0x', '') for c in parts[0]])
+                number = MagicNumber(bytes, parts[1], parts[2])
+                self.add(number)
+
+
+    def add(self, number: MagicNumber):
```

Comment:
Should this be a "private" method?
```suggestion
    def _add(self, number: MagicNumber):
```